src/QirRuntime/test/QIR-static/compiler/QirTarget.qs

```diff
@@ -1,135 +1,63 @@
 
     open Microsoft.Quantum.Targeting;
-    open Microsoft.Quantum.Instructions as Phys;
-    
-    @Inline()
-    function PI() : Double
-    {
-        return 3.14159265357989;
-    }
-    
-    function IntAsDouble(i : Int) : Double {
+
+    operation X(qb : Qubit) : Unit
+    is Adj + Ctl {
         body intrinsic;
+        adjoint self;
     }
 
-    operation X(qb : Qubit) : Unit
-    is Adj {
+    operation Y(qb : Qubit) : Unit
+    is Adj + Ctl {
         body intrinsic;
         adjoint self;
-	}
+    }
 
     operation Z(qb : Qubit) : Unit
-    is Adj {
+    is Adj + Ctl {
         body intrinsic;
         adjoint self;
-	}
+    }
 
     operation H(qb : Qubit) : Unit
-    is Adj {
+    is Adj + Ctl {
         body intrinsic;
         adjoint self;
-	}
-
-    operation T(qb : Qubit) : Unit 
-    is Adj {
-        body intrinsic;
-	}
+    }
 
-    operation CNOT(control : Qubit, target : Qubit) : Unit
-    is Adj {
+    operation S(qb : Qubit) : Unit
+    is Adj + Ctl {
         body intrinsic;
-        adjoint self;
-	}
+    }
 
-    @TargetInstruction("mz")
-    operation M(qb : Qubit) : Result {
+    operation T(qb : Qubit) : Unit
+    is Adj + Ctl {
         body intrinsic;
-	}
+    }
 
-    operation Measure(bases : Pauli[], qubits : Qubit[]) : Result {
+    operation R (pauli : Pauli, theta : Double, qubit : Qubit) : Unit
+    is Adj + Ctl
+    {
         body intrinsic;
     }
 
-    operation MResetZ(qb : Qubit) : Result
+    operation Exp (paulis : Pauli[], theta : Double, qubits : Qubit[]) : Unit
+    is Adj + Ctl
     {
-        let res = M(qb);
-        if (res == One)
-        {
-            X(qb);
-        }
-        return res;
+        body intrinsic;
     }
 
-    @Inline()
-    operation S(qb : Qubit) : Unit
-    is Adj {
-        body  (...)
-        {
-            Phys.S(qb);  
-		}
-        adjoint (...)
-        {
-            Phys.S(qb);
-            Z(qb);
-        }
-	}
-
-    @Inline()
-    operation Rx(theta : Double, qb : Qubit) : Unit
-    is Adj {
-        body  (...)
-        {
-            Phys.Rx(theta, qb);  
-		}
-        adjoint (...)
-        {
-            Phys.Rx(-theta, qb);  
-		}
-	}
+    operation Measure(bases : Pauli[], qubits : Qubit[]) : Result {
+        body intrinsic;
+    }
 
-    @Inline()
-    operation Rz(theta : Double, qb : Qubit) : Unit
-    is Adj + Ctl {
-        body  (...)
-        {
-            Phys.Rz(theta, qb);  
-		}
-        adjoint (...)
-        {
-            Phys.Rz(-theta, qb);  
-		}
-        controlled (ctls, ...)
-        {
-            Phys.Rz(theta / 2.0, qb);
-            CNOT(ctls[0], qb);
-            Phys.Rz(-theta / 2.0, qb);
-            CNOT(ctls[0], qb);
-        }
-        controlled adjoint (ctls, ...)
-        {
-            Phys.Rz(-theta / 2.0, qb);
-            CNOT(ctls[0], qb);
-            Phys.Rz(theta / 2.0, qb);
-            CNOT(ctls[0], qb);
-        }
-	}
+    // Non-standard operation. Should it exist?
```

Comment:
In my opinion, no, it can be removed. But that may be better handled in a separate clean-up PR later if there are test changes that need to be made too?

Reply:
Yep, after talking to Guen and seeing the confusion this callback creates, I went ahead and removed it.